app/Http/Controllers/SaveController.php

```diff
@@ -61,14 +82,25 @@ public function store(Request $request): JsonResponse
             "name" => "required|string|max:255",
             "description" => "string|max:300",
             "data" => "nullable|json",
-            "tool_id" => "required|exists:tools,id"
+            "tool_id" => "required|exists:tools,id",
+            "resources.*" => [
+                File::types(self::ALLOWED_MIMETYPES)
+                    ->max(self::FILE_MAX_KILOBYTES)
+            ]
         ]);
+        $s = DB::transaction(function () use ($request, $validate) {
+            $s = new Save($validate);
+            $s->tool_id = $validate["tool_id"];
+            $s->owner_id = $request->user()->id;
+            $s->save();
+
+            if (array_key_exists("resources", $validate)) {
+                $this->resourceService->saveResources($s, $validate["resources"]);
```

Comment:
Doesnt work. in the $validated the resources are apparently not listed. to access files one needs to do `$request->allFiles();`

_Added changes in new commit._

Reply:
If the frontend is changed described [here](https://github.com/ricom/toolbox-frontend/pull/92#discussion_r1348090342) this validation would work.